File: app/src/main/java/io/treehouses/remote/bases/BaseDialogFragment.java
package io.treehouses.remote.bases;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import io.treehouses.remote.Network.BluetoothChatService;
import io.treehouses.remote.callback.HomeInteractListener;

public class BaseDialogFragment extends DialogFragment {
    public HomeInteractListener listener;
    public BluetoothChatService mChatService;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        if (context instanceof HomeInteractListener)
            listener = (HomeInteractListener) context;
        else
            throw new RuntimeException("Implement interface first");

        mChatService = listener.getChatService();
    }
}

File: app/src/main/kotlin/io/treehouses/remote/bases/BaseDialogFragment.kt
package io.treehouses.remote.bases

import android.content.Context
import androidx.fragment.app.DialogFragment
import io.treehouses.remote.callback.HomeInteractListener

open class BaseDialogFragment : DialogFragment() {
    @JvmField
    var listener: HomeInteractListener? = null
    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is HomeInteractListener) listener = context
    }
}